src/main/java/com/yahoo/memory/BaseWritableMemoryImpl.java

```diff
@@ -58,64 +62,101 @@
     this.localReadOnly = localReadOnly;
   }
 
+  static BaseWritableMemoryImpl newInstance(final ResourceState state,
+      final boolean localReadOnly) {
+    if (state.getCapacity() == 0) { return WritableMemoryImpl.ZERO_SIZE_MEMORY; }
```

Comment:
The static field BaseWritableMemoryImpl.ZERO_SIZE_MEMORY should be accessed directly.